app/controllers/api/articles_controller.rb

```diff
@@ -10,6 +11,9 @@ def index
 
   def create
     article = Article.create(article_params)
+    article.category_id = if Category.find_by(name: article.category_name)
+                            Category.find_by(name: article.category_name).id
+                          end
```

Comment:
Some code repetition here. You can use safe navigation operator `&.` to get rid of it, It is analog of `?.` in java script.
So instead of having `if` statement you can do `Category.find_by(name: article.category_name)&.id` that should do the trick.

Reply:
This is amazing! It looks so much cleaner and works like a charm